src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -501,9 +567,13 @@ public Solution RemoveProjectReference(ProjectId projectId, ProjectReference pro
         /// Create a new solution instance with the project specified updated to contain
         /// the specified list of project references.
         /// </summary>
-        public Solution WithProjectReferences(ProjectId projectId, IEnumerable<ProjectReference> projectReferences)
+        /// <param name="projectId">Id of the project whose references to replace with <paramref name="projectReferences"/>.</param>
+        /// <param name="projectReferences">New project references. May include duplicate entries.</param>
```

Comment:
Doesn't ToBoxedImmutableArrayWithDistinctNonNullItems *throw* if there is a duplicate?

Reply:
Ah, yes, I changed it.